components/dialog/README.md

```diff
@@ -0,0 +1,70 @@
+# Dialogs
+
+## d2l-dialog
+
+The `d2l-dialog` element is a generic dialog that provides a slot for arbitrary content, and a `footer` slot for workflow buttons.
+
+![Dialog](./screenshots/dialog.png?raw=true)
+
+```html
+<script type="module">
+  import '@brightspace-ui/core/components/dialog/dialog.js';
+</script>
+
+<d2l-button id="open">Show Dialog</d2l-button>
+
+<d2l-dialog title-text="Dialog Title">
+  <div>Some dialog content</div>
+  <d2l-button slot="footer" primary>Done</d2l-button>
+  <d2l-button slot="footer" id="cancel">Cancel</d2l-button>
+</d2l-dialog>
+
+<script>
+  document.querySelector('#open').addEventListener('click', () => {
+    document.querySelector('d2l-dialog').opened = true;
+  });
+  document.querySelector('#cancel').addEventListener('click', () => {
+    document.querySelector('d2l-dialog').opened = false;
+  });
+</script>
+```
+
+**Properties:**
+
+- `opened`: Whether or not the dialog is open
```

Comment:
I've been putting the type and whether it's optional here in brackets after the name. E.g. "`title-text` (String, required): Text diplayed..."